src/crypto/crypto_x509.cc

```diff
@@ -58,246 +63,222 @@ void Fingerprint(const FunctionCallbackInfo<Value>& args) {
-void X509Certificate::SubjectAltName(const FunctionCallbackInfo<Value>& args) {
-  ReturnPropertyThroughBIO<GetSubjectAltNameString>(args);
+void SerialNumber(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  X509Certificate* cert;
+  ASSIGN_OR_RETURN_UNWRAP(&cert, args.This());
+  auto serial = cert->view().getSerialNumber();
+  if (serial) {
+    args.GetReturnValue().Set(OneByteString(
+        env->isolate(), static_cast<unsigned char*>(serial.get())));
+  }
 }
 
-void X509Certificate::InfoAccess(const FunctionCallbackInfo<Value>& args) {
-  ReturnPropertyThroughBIO<GetInfoAccessString>(args);
-}
+void PublicKey(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  X509Certificate* cert;
+  ASSIGN_OR_RETURN_UNWRAP(&cert, args.This());
 
-void X509Certificate::ValidFrom(const FunctionCallbackInfo<Value>& args) {
-  ReturnPropertyThroughBIO<GetValidFrom>(args);
-}
+  // TODO(tniessen): consider checking X509_get_pubkey() when the
+  // X509Certificate object is being created.
+  auto result = cert->view().getPublicKey();
+  if (!result.value) {
+    ThrowCryptoError(env, result.error.value_or(0));
+    return;
+  }
+  std::shared_ptr<KeyObjectData> key_data = KeyObjectData::CreateAsymmetric(
+      kKeyTypePublic, ManagedEVPPKey(std::move(result.value)));
 
-void X509Certificate::ValidTo(const FunctionCallbackInfo<Value>& args) {
-  ReturnPropertyThroughBIO<GetValidTo>(args);
+  Local<Value> ret;
+  if (KeyObjectHandle::Create(env, std::move(key_data)).ToLocal(&ret)) {
+    args.GetReturnValue().Set(ret);
+  }
 }
 
-template <MaybeLocal<Value> Property(Environment* env, X509* cert)>
-static void ReturnProperty(const FunctionCallbackInfo<Value>& args) {
+void KeyUsage(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
   X509Certificate* cert;
   ASSIGN_OR_RETURN_UNWRAP(&cert, args.This());
-  Local<Value> ret;
-  if (Property(env, cert->get()).ToLocal(&ret)) args.GetReturnValue().Set(ret);
-}
 
-void X509Certificate::KeyUsage(const FunctionCallbackInfo<Value>& args) {
-  ReturnProperty<GetKeyUsage>(args);
-}
+  auto eku = cert->view().getKeyUsage();
+  if (!eku) return;
+
+  const int count = sk_ASN1_OBJECT_num(eku.get());
+  MaybeStackBuffer<Local<Value>, 16> ext_key_usage(count);
+  char buf[256];
+
+  int j = 0;
+  for (int i = 0; i < count; i++) {
+    if (OBJ_obj2txt(buf, sizeof(buf), sk_ASN1_OBJECT_value(eku.get(), i), 1) >=
+        0) {
+      ext_key_usage[j++] = OneByteString(env->isolate(), buf);
+    }
+  }
 
-void X509Certificate::SerialNumber(const FunctionCallbackInfo<Value>& args) {
-  ReturnProperty<GetSerialNumber>(args);
+  args.GetReturnValue().Set(
+      Array::New(env->isolate(), ext_key_usage.out(), count));
 }
 
-void X509Certificate::Raw(const FunctionCallbackInfo<Value>& args) {
-  ReturnProperty<GetRawDERCertificate>(args);
+void CheckCA(const FunctionCallbackInfo<Value>& args) {
```

Comment:
Can you add a TODO here to add a V8 Fast API version for this, since it returns a boolean?

Reply:
I plan to revisit the entire api to see where we can add fast api calls throughout. Mind if I add this later? This change set is already large and want to avoid too much extraneous line changes

Reply:
Yeah, no worries. Just FYI. It might be good to have these in a list somewhere, for new contributors to help. I know @tniessen would be interested (like me) to add some fast  api calls for these.